crates/task-impls/src/da.rs

```diff
@@ -244,15 +239,19 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                 // there is already a view there: the replica task may have inserted a `Leaf` view which
                 // contains strictly more information.
                 consensus.state_map.entry(view).or_insert(View {
-                    view_inner: ViewInner::DA {
-                        block: payload_commitment,
-                    },
+                    view_inner: ViewInner::DA { payload_commitment },
                 });
 
                 // Record the block payload we have promised to make available.
-                consensus
+                if let Err(e) = consensus
                     .saved_block_payloads
-                    .insert(proposal.data.block_payload);
+                    .insert(BlockPayload::from_bytes(
```

Comment:
This is what Jeb mentioned in another comment. We are converting bytes to payload, then we convert payload to bytes at the other side. Maybe instead just stay with bytes and save all the conversion.